Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/RoundTransform.java

```diff
@@ -0,0 +1,43 @@
+package org.dolphinemu.dolphinemu.utils;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapShader;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.RectF;
+import android.graphics.Shader;
+
+public class RoundTransform implements com.squareup.picasso.Transformation {
+
+  private final int cornerRadius;
+
+  // Radius is corner rounding in dp
+  public RoundTransform(int cornerRadius) {
+    this.cornerRadius = cornerRadius;
+  }
+
+  @Override
+  public Bitmap transform(Bitmap source) {
+    final Paint paint = new Paint();
+    paint.setAntiAlias(true);
+    paint.setShader(new BitmapShader(source, Shader.TileMode.CLAMP,
+      Shader.TileMode.CLAMP));
+
+    Bitmap output = Bitmap.createBitmap(source.getWidth(),
+      source.getHeight(), Bitmap.Config.ARGB_8888);
+    Canvas canvas = new Canvas(output);
+    canvas.drawRoundRect(new RectF(0, 0, source.getWidth(),
+      source.getHeight()), cornerRadius, cornerRadius, paint);
+
+    if (source != output)
```

Comment:
Is there ever a case where these will be the same object?

Reply:
I thought that the changes would be saved as part of the cached image on disk. Turns out that is not the case because I made an error earlier that led me to believe that. I will recycle the source every time regardless now and the CORNER_RADIUS variable will be applied at runtime, not cached. My bad.